### PR TITLE
Use rtk hooks for public questions

### PR DESCRIPTION
**Pull request by oisincoveney**
Removes the actions for `CREATE_PUBLIC_LINK` and `DELETE_PUBLIC_LINK`, and modifies the RTK endpoints to include IDs. Then, I've removed the intermediate actions and use the RTK query hooks directly in the redux actions with matchers.

### CODE REVIEW

File: frontend/src/metabase/query_builder/reducers-typed.ts
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/reducers-typed.ts:1]: this diff is being weird - for example line 111 and line 110 on the left and right side respectively aren't reflective of the changes I made. I removed the create/delete public link actions, and moved them to matchers at the bottom.

File: frontend/src/metabase/sharing/components/PublicLinkPopover/QuestionPublicLinkPopover.tsx [hunk 1]
```diff
@@ -42,15 +39,18 @@ export const QuestionPublicLinkPopover = ({
...
+    await createPublicQuestionLink({ id: question.id() });
   };
-  const deletePublicQuestionLink = async () => {
+  const handleDeletePublicQuestionLink = async () => {
     trackPublicLinkRemoved({
       artifact: "question",
       source: "public-share",
     });
-    await dispatch(deletePublicLink(question.card()));
+    await deletePublicQuestionLink({ id: question.id() });
```
Comment (heypoom) [inline frontend/src/metabase/sharing/components/PublicLinkPopover/QuestionPublicLinkPopover.tsx:53]: (Just noticed that we bring in the entire card just to delete a public link in the old code, wow.)

**Review by heypoom — APPROVED** [review]
Nice work!

File: frontend/src/metabase/sharing/components/PublicLinkPopover/QuestionPublicLinkPopover.tsx (hunk 1 above)
Comment (oisincoveney) [reply frontend/src/metabase/sharing/components/PublicLinkPopover/QuestionPublicLinkPopover.tsx:53]: Yeah it wasn't the best idea 😂

**Pull request merged**